dashboard/test/mailers/ops_mailer_test.rb

```diff
@@ -15,4 +15,114 @@ class OpsMailerTest < ActionMailer::TestCase
     assert_match "added 1 teachers", mail.body.encoded
   end
 
+  test "workshop_in_2_weeks_reminder" do
+    # Create a workshop that starts 14 days from now. Include other segments for authenticity
+    @workshop_1 = create(:workshop, phase: 2, cohorts: [create(:cohort, id: 1,
+      teachers: [create(:teacher, email: 'abc@code.org'), create(:teacher, email: 'xyz@code.org')])],
+      facilitators: [create(:facilitator, email: 'facilitator_1@code.org')])
+    @workshop_1.segments.clear
+    @workshop_1.segments << create(:segment, workshop_id: @workshop_1.id, start: DateTime.now + 14.day, end: DateTime.now + 14.day + 60.minute)
+    @workshop_1.segments << create(:segment, workshop_id: @workshop_1.id, start: DateTime.now + 15.day, end: DateTime.now + 15.day + 60.minute)
+    @workshop_1.segments << create(:segment, workshop_id: @workshop_1.id, start: DateTime.now + 16.day, end: DateTime.now + 16.day + 60.minute)
+    # Create a workshop that does not start 14 days from now. Include one segment that is 14 days from now to test
+    # that it doesn't get confused about the actual start of the workshop
+    @workshop_2 = create(:workshop, phase: 8, cohorts: [create(:cohort, id: 2,
+      teachers: [create(:teacher, email: '123@code.org'), create(:teacher, email: '987@code.org')])],
+      facilitators: [create(:facilitator, email: 'facilitator_2@code.org')])
+    @workshop_2.segments << create(:segment, workshop_id: @workshop_2.id, start: DateTime.now + 13.day, end: DateTime.now + 13.day + 60.minute)
+    @workshop_2.segments << create(:segment, workshop_id: @workshop_2.id, start: DateTime.now + 14.day, end: DateTime.now + 14.day + 60.minute)
+    @workshop_2.segments << create(:segment, workshop_id: @workshop_2.id, start: DateTime.now + 15.day, end: DateTime.now + 15.day + 60.minute)
+
```

Comment:
It would also be good to test the case where the workshop is entirely before or entirely after 14 days from now.

You might want to add a create_workshop help function to cut down on the repeated code.